From d176c94e8095656672f33dcccb4059bc3ea99fa7 Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Fri, 16 Jun 2017 14:41:31 -0300
Subject: [PATCH 1/4] Initilize IDE prior environment
Organization: O.S. Systems Software LTDA.

To allow the loading of environment from IDE (for example, using FAT)
the initilization of IDE subsystem must come before the environment
one.

Successfully tested on QEMU x86.

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---
 common/board_r.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/common/board_r.c b/common/board_r.c
index d69a33c4a3..04a2ea3561 100644
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -818,6 +818,9 @@ static init_fnc_t init_sequence_r[] = {
 #ifdef CONFIG_GENERIC_MMC
 	initr_mmc,
 #endif
+#if defined(CONFIG_CMD_IDE)
+	initr_ide,
+#endif
 #ifdef CONFIG_HAS_DATAFLASH
 	initr_dataflash,
 #endif
@@ -890,9 +893,6 @@ static init_fnc_t init_sequence_r[] = {
 #if defined(CONFIG_CMD_PCMCIA) && !defined(CONFIG_CMD_IDE)
 	initr_pcmcia,
 #endif
-#if defined(CONFIG_CMD_IDE)
-	initr_ide,
-#endif
 #ifdef CONFIG_LAST_STAGE_INIT
 	INIT_FUNC_WATCHDOG_RESET
 	/*
-- 
2.13.1

